code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -525,6 +525,17 @@
 		desc = "Your standard drinking glass."
 		return
 
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(!user.canUseTopic(src, be_close=TRUE))
+		return
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
Ghosts and other people not close to the drinking glass can still use it.

- Use a pen on a drink.
- Keep the input screen open, and walk away!
- Watch as you magically rename a drink when not close to it! WOAH!

This can be solved by simply running `canUseTopic` before `if(txt)` and after `var/txt` to ensure the user is still there when the `stripped_input` proc completes.